### PR TITLE
TSK-1349: Unit tests for workbasket-details, -information

### PR DESCRIPTION
**Pull request by cnguyen-de**
https://taskana.atlassian.net/browse/TSK-1349

### CODE REVIEW

**Comment by gitgoodjhe** [comment]
This PR contains 13 commits, could you please squash them? https://taskana.atlassian.net/wiki/spaces/TAS/pages/228884481/GIT+-+basic+Workflow+Usage under the section !squash commits"

**Comment by cnguyen-de** [comment]
Commits will be squashed during merge

File: web/src/app/administration/components/workbasket-details/workbasket-details.component.ts [hunk 1]
```diff
@@ -108,12 +97,6 @@ export class WorkbasketDetailsComponent implements OnInit, OnDestroy {
     if (!workbasketIdSelected && this.action === ACTION.CREATE) {
       // CREATE
       this.workbasket = {};
-      this.domainService
-        .getSelectedDomain()
-        .pipe(takeUntil(this.destroy$))
-        .subscribe((domain) => {
-          this.workbasket.domain = domain;
-        });
```
Comment (sofie29) [inline web/src/app/administration/components/workbasket-details/workbasket-details.component.ts:116]: Why do we not need this anymore?

File: web/src/app/administration/components/workbasket-details/workbasket-details.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    store = TestBed.inject(Store);
+    actions$ = TestBed.inject(Actions);
+    fixture.detectChanges();
+  }));
+
+  it('should create component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should display loading spinner while content loads', () => {
```
Comment (sofie29) [inline web/src/app/administration/components/workbasket-details/workbasket-details.component.spec.ts:97]: This is not our naming convention

File: web/src/app/administration/components/workbasket-details/workbasket-details.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    store.reset({
+      ...store.snapshot(),
+      workbasket: workbasketCopyState
+    });
+    component.ngOnInit();
+    fixture.detectChanges();
+    expect(component.workbasketCopy).toEqual(component.workbasket);
+  });
+
+  it('should render copied workbasket when action is READ', () => {
```
Comment (sofie29) [inline web/src/app/administration/components/workbasket-details/workbasket-details.component.spec.ts:147]: I think you didn't mean '**copied** workbasket' 😀

**Review by sofie29 — CHANGES_REQUESTED** [review]

File: web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+    fixture = TestBed.createComponent(WorkbasketInformationComponent);
+    debugElement = fixture.debugElement;
+    component = fixture.componentInstance;
+    store = TestBed.inject(Store);
+    actions$ = TestBed.inject(Actions);
+    store.reset({
+      ...store.snapshot(),
+      engineConfiguration: engineConfigurationMock,
+      workbasket: workbasketReadState
```
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts:134]: yes

[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts:134]: I'll rename them

File: web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+  it('should display copy badge message when action is COPY', () => {
+    component.action = ACTION.COPY;
+    component.ngOnChanges();
+    expect(component.badgeMessage).toContain(`Copying workbasket: ${component.workbasket.key}`);
+  });
+
+  it('should set type variable in selectType', () => {
+    const type = ICONTYPES.GROUP;
+    component.selectType(type);
+    expect(component.workbasket.type).toMatch(type);
```
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts:171]: just want to have 100% in jest coverage although this test doesn't matter much

File: web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+  });
+
+  it('should submit if validatorService is true', () => {
```
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts:174]: corrected

File: web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+  it('should reset workbasket information when onUndo is called', () => {
+    component.workbasketClone = selectedWorkbasket;
+    const notificationService = TestBed.inject(NotificationService);
+    const toastSpy = jest.spyOn(notificationService, 'showToast');
+    component.onUndo();
+    expect(toastSpy).toHaveBeenCalled();
+    expect(component.workbasket).toMatchObject(component.workbasketClone);
+  });
+
+  it('should save workbasket if workbasketId there', () => {
```
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts:189]: corrected

File: web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+    component.workbasket = { ...selectedWorkbasket };
+    component.workbasket.workbasketId = '1';
+    component.action = ACTION.COPY;
+    let actionDispatched = false;
+    actions$.pipe(ofActionDispatched(UpdateWorkbasket)).subscribe(() => (actionDispatched = true));
```
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts:194]: corrected

File: web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+  });
+
+  it('should dispatch MarkWorkbasketforDeletion action when onRemoveConfirmed is called', () => {
+    let actionDispatched = false;
+    actions$.pipe(ofActionDispatched(MarkWorkbasketForDeletion)).subscribe(() => (actionDispatched = true));
+    component.onRemoveConfirmed();
+    expect(actionDispatched).toBe(true);
+  });
+
+  it('should create new workbasket if workbasketId is undefined', () => {
```
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/workbasket-information/workbasket-information.component.spec.ts:207]: corrected

File: web/src/app/administration/components/workbasket-details/workbasket-details.component.ts (hunk 1 above)
Comment (cnguyen-de) [reply web/src/app/administration/components/workbasket-details/workbasket-details.component.ts:116]: I'm gonna restore this part, will need to discuss together if we should remove it

**Review by sofie29 — APPROVED** [review]

**Pull request merged**